src/components/forms/RoutingTableForm.js

```diff
@@ -0,0 +1,182 @@
+import { IcseSelect } from "../Dropdowns";
+import { IcseNameInput, IcseToggle } from "../Inputs";
+import { IcseFormGroup } from "../Utils";
+import RoutingTableRouteForm from "./RoutingTableRouteForm";
+import IcseFormTemplate from "../IcseFormTemplate";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import PropTypes from "prop-types";
+import { transpose, isNullOrEmptyString } from "lazy-z";
+
+class RoutingTableForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...this.props.data };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+    this.handleToggle = this.handleToggle.bind(this);
+  }
+
+  /**
+   * handle input change
+   * @param {string} name key to change in state
+   * @param {*} value value to update
+   */
+  handleInputChange(event) {
+    let nextState = { ...this.state };
+    let { name, value } = event.target;
+    nextState[name] = value;
+    this.setState(nextState);
+  }
+
+  handleToggle(name) {
+    this.setState(this.toggleStateBoolean(name, this.state));
+  }
+  render() {
+    let composedId = this.props.data.name + "-route-form";
+    let innerFormProps = {
+      arrayParentName: this.props.data.name,
+      cluster: this.props.data,
+      invalidTextCallback: this.props.invalidRouteTextCallback,
+      invalidCallback: this.props.invalidRouteCallback,
+    };
+    transpose({ ...this.props.routeProps }, innerFormProps);
+    return (
+      <>
+        <IcseFormGroup>
+          <IcseNameInput
+            componentName="routing-table-route"
+            id={this.props.data.name + "-route-name"}
+            hideHelperText
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            invalidCallback={() => {
+              this.props.invalidCallback(this.state, this.props);
+            }}
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+          />
+          <IcseSelect
+            formName="vsi_form"
+            name="vpc"
+            labelText="VPC"
+            groups={this.props.vpcList}
+            value={this.state.vpc}
+            handleInputChange={this.handleInputChange}
+            invalid={isNullOrEmptyString(this.state.vpc)}
+            invalidText="Select a VPC."
+          />
+        </IcseFormGroup>
+        <IcseFormGroup>
+          <IcseToggle
+            id={composedId + "-direct-link-toggle"}
+            labelText="Direct Link Ingress"
+            defaultToggled={this.state.route_direct_link_ingress}
+            onToggle={this.handleToggle}
+            tooltip={{
+              content:
+                "If set to true, the routing table is used to route traffic that originates from Direct Link to the VPC. To succeed, the VPC must not already have a routing table with the property set to true",
+            }}
```

Comment:
might need to add `align` and `alignModal` values so tooltip doesn't get cut off

Reply:
what alignment ?

Reply:
`bottom-left` for both I believe